### DISCUSSION TITLE
values with defaultValues

### DISCUSSION THREAD

**Discussion by pavelspichonak** [react-hook-form/react-hook-form#3789] [Q&A]
@bluebill1049 now we can pass `defaultValues` to `useForm` hook and after that we can use `isDirty` for example to disable submit button if there are no changes.

But imagine case when we have modal window which contains form. When modal window is closed and opened again I'd like to restore previously added form values by user. Now it looks like I can do it only using `defaultValues` but in this case `isDirty` value will be `false` but it should be `true` because user updated form values when he opened modal window first time.

Maybe it is possible to handle somehow such case? If there is no API to handle this maybe you can add `values` property to `useForm` hook to be able to restore previously added form values by user and in this case `isDirty` will be `true` because passed `values` differ from `defaultValues`? Thanks.

**Comment by markivancho** [comment]
Reset should help you - https://react-hook-form.com/api/#reset
You will be able to inject new `defaultValues` and still have `isDirty` unchanged

**Comment by pavelspichonak** [comment]
@markivancho as far as I understand `reset` sets new default values so after resetting `isDirty` will be `true` only when form values differ from new `defaultValues` passed to reset function. But it is already not that `defaultValues` which form had when you open Modal window containing form first time. @bluebill1049 am I right?

**Reply by bluebill1049** [reply]
I think we will introduce omitResetState with `values`, so you only skip reset `defaultValues`. Is this what you after?

**Reply by pavelspichonak** [reply]
If it allows to set form `values` and does not affect `defaultValues` - it would be great.

**Reply by bluebill1049** [reply]
That's it. This will be in the next minor update.

**Reply by pavelspichonak** [reply]
thanks a lot!

**Reply by bluebill1049** [reply]
@pavelspichonak I think setValue will that for you right?

**Reply by pavelspichonak** [reply]
@bluebill1049 I'm not sure but when I used `formik` before and tried to use something like `setValue` in your library it worked so slowly when you have huge form with lots of inputs. So I believe it will work slowly if we use `setValue` from `react-hook-form` library. But I'm not sure.

**Reply by pavelspichonak** [reply]
I believe it is much better to be able to set all form values at once instead of calling `setValue` for each property from `values`

**Reply by bluebill1049** [reply]
Try it with `setValue` first, maybe it's better.
